common/src/main/java/io/crate/interval/PGIntervalParser.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.interval;
+
+import org.joda.time.Period;
+
+import java.util.StringTokenizer;
+
+import static io.crate.interval.IntervalParser.nullSafeIntGet;
+import static io.crate.interval.IntervalParser.parseInteger;
+import static io.crate.interval.IntervalParser.parseMilliSeconds;
+import static io.crate.interval.IntervalParser.roundToPrecision;
+
+class PGIntervalParser {
+
+    static Period apply(String value, IntervalParser.Precision start, IntervalParser.Precision end) {
+        return roundToPrecision(apply(value), start, end);
+    }
+
+    static Period apply(String value) {
```

Comment:
I think the methods here could also use some docs to explain the format of the strings they're parsing.